src/invidious/routes/api/v1/authenticated.cr

```diff
@@ -22,12 +22,11 @@ module Invidious::Routes::API::V1::Authenticated
     user = env.get("user").as(User)
 
     begin
-      preferences = Preferences.from_json(env.request.body || "{}")
+      user.preferences = Preferences.from_json(env.request.body || "{}")
     rescue
-      preferences = user.preferences
     end
 
-    PG_DB.exec("UPDATE users SET preferences = $1 WHERE email = $2", preferences.to_json, user.email)
+    Invidious::Database::Users.update_preferences(user)
```

Comment:
I believe this is the first update method I've seen that takes in the object instead of the fields being updated

Reply:
Probably ^^ I've only done that for a handful of situations, when it didn't require to dig too much in the logic of the code around. I've planned to make another PR after this one to simplify the DB calls (mostly duplicates with very small variations).